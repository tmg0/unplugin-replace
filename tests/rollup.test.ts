import path from 'node:path'
import { expect, test } from 'vitest'
import { rollupBuild } from '@vue-macros/test-utils'
import UnpluginReplace from '../src/rollup'

test('rollup', async () => {
  const result = await rollupBuild(
    path.resolve(__dirname, 'fixtures/main.js'),
    [
      UnpluginReplace({
        'process.platform': '"darwin"',
      }),
    ],
  )
  expect(result).toMatchSnapshot()
})
