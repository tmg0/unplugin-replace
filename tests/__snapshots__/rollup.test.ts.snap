// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`rollup 1`] = `
"import 'node:process';

const platform = "darwin";

console.log('hello "darwin"', platform);

export { platform };
"
`;
